abel/tools/circularize.py

```diff
@@ -0,0 +1,236 @@
+# -*- coding: utf-8 -*-
+from __future__ import division
+import numpy as np
+from scipy import ndimage
+from scipy.interpolate import UnivariateSpline
+from scipy.optimize import leastsq
+
+import abel
+
+# see https://github.com/PyAbel/PyAbel/issues/186 for discussion
+
+
+def circularize_image(IM, method="lsq", center=None, radial_range=None,
+                      zoom=1, smooth=0, nslices=32, ref_angle=None, 
+                      inverse=False, return_correction=False):
+    """
+    Remove radial distortion from a velocity-map-image through radial scaling
```

Comment:
I don't think that anything about the image circularization is specific to VMI images. Instead, this circularization can be applied to any image that has features that exist at a constant radius. So, perhaps the summary is something like: "This function corrects for radial distortion in an image that consists of features that exist at a constant radius. In other words, the image must consist of "circle-like" structures. This function attempts to correct for any distortion and ensures that the circle-like structures are perfect circles. 

This function is especially useful for correcting the image obtained with a velocity-map-imaging spectrometer with imperfect electro-static lenses, allowing a high-resolution 1D photoelectron distribution to be extracted. 

The algorithm is to split the image into "slices" at many different angles (set by `nslices`) and compare adjacent slices. A scaling factor is found which aligns each slice with the previous slice. The image is then corrected using a spline function that smoothly connects the discrete scaling factors as a continuous function of angle."